src/app/core/cookie-agreement.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CookieAgreementService {
+  cookiesAgreement = false;
+  showCookieBanner = true;
+
+  constructor() {}
+
+  getCookieAgreementVal(): boolean {
+    this.checkCookieAgreement();
+    return this.cookiesAgreement;
+  }
+
+  getShowCookieBannerVal(): boolean {
+    this.checkShowCookieBanner();
+    return this.showCookieBanner;
+  }
+
+  /**
+   * @Check_Functions
+   * checks local storage for value of
+   */
+  private checkCookieAgreement() {
+    const agreement = localStorage.getItem('acceptCookieAgreement');
+    this.cookiesAgreement = agreement !== null && agreement === 'true';
+  }
+  private checkShowCookieBanner() {
+    const agreement = localStorage.getItem('showCookieBanner');
+    this.showCookieBanner = !(agreement !== null && agreement === 'false');
```

Comment:
Since this in production, what if I have accepted the cookie agreement in the past? Will there be any issue with me logging in? If there is we should make sure to set the show cookieBanner appropriately. We haven't updated our terms of service so if they have already accepted it we can allow them to login.

Reply:
@pzalep1 I did update the terms of service to add the cookies paragraph similar to CARD. With current implementation there will be a problem requiring them to accept again.  However, this can be avoided by changing  acceptCookieAgreement (new) to cookieAgreement (currently) as the localStorage value to look for.

Reply:
That sounds good. By updated I we didn't "add" anything new we just added further explanation on our cookies.

Reply:
Ok sounds good. I just added logic and tested it. Anyone who has previously accepted the cookies will not be requested to accept again. This works for both the banner and the popup. @pzalep1